src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -12,11 +20,49 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Phone create(Phone phone) {
-        return null;
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(phone);
+            transaction.commit();
+            return phone;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert a phone " + phone, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Phone> findAll(Map<String, String[]> params) {
-        return null;
+        Session session = factory.openSession();
+        if (params != null && !params.isEmpty()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Phone> query = criteriaBuilder.createQuery(Phone.class);
+            Root<Phone> root = query.from(Phone.class);
+            // Predicates
+            Predicate predicate = null;
+            Set<String> keys = params.keySet();
+            for (String key : keys) {
+                CriteriaBuilder.In<String> tempPredicate = criteriaBuilder.in(root.get(key));
+                Arrays.stream(params.get(key)).forEach(tempPredicate::value);
+                if (predicate == null) {
+                    predicate = tempPredicate;
+                } else {
+                    predicate = criteriaBuilder.and(predicate, tempPredicate);
+                }
+            }
+            query.where(predicate);
+            return session.createQuery(query).getResultList();
+        } else {
+            return session.createQuery("from Phone", Phone.class).getResultList();
+        }
```

Comment:
```suggestion
        } 
       return session.createQuery("from Phone", Phone.class).getResultList();
```
IMHO - it's better to avoid if else blocks for sake of simplicity, if there is return in at end, else is redundant